### PR TITLE
Render reflectection

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/angular2/src/facade/collection.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class MapWrapper {
   static clearValues(m: Map<any, any>) { _clearValues(m); }
   static iterable(m) { return m; }
   static keys<K>(m: Map<K, any>): List<K> { return m.keys(); }
```
Comment (tbosch) [inline modules/angular2/src/facade/collection.ts]: Also fix `Map.keys`

File: modules/angular2/src/facade/collection.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ export class MapWrapper {
...
-  static values<V>(m: Map<any, V>): List<V> { return m.values(); }
+  static values<V>(m: Map<any, V>): List<V> { return Array['from'](m.values()); }
```
Comment (tbosch) [inline modules/angular2/src/facade/collection.ts]: Maybe rather use `<any>`

File: modules/angular2/src/render/api.ts [hunk 3]
```diff
@@ -358,19 +371,33 @@ export class Renderer {
...
+  /**
+   * Sets a class on an element.
+   */
+  setElementClass(viewRef: RenderViewRef, elementIndex: number, className: string, isAdd: boolean) {
+  }
+
+  /**
+   * Sets a style on an element.
+   */
+  setElementStyle(viewRef: RenderViewRef, elementIndex: number, styleName: string,
```
Comment (tbosch) [inline modules/angular2/src/render/api.ts]: Maybe remove `unit` here and concatenate on app side?

File: modules/angular2/src/render/api.ts [hunk 4]
```diff
@@ -358,19 +371,33 @@ export class Renderer {
...
-   * in the ProtoView
+   * Sets an attribute on an element.
+   */
+  setElementAttribute(viewRef: RenderViewRef, elementIndex: number, attributeName: string,
+                      attributeValue: string) {}
+
+  /**
+   * Sets a class on an element.
+   */
+  setElementClass(viewRef: RenderViewRef, elementIndex: number, className: string, isAdd: boolean) {
```
Comment (tbosch) [inline modules/angular2/src/render/api.ts]: Rename to `toggleElementClass`

File: modules/angular2/src/render/dom/compiler/compiler.ts [hunk 5]
```diff
@@ -17,7 +17,6 @@ import {TemplateLoader} from 'angular2/src/render/dom/compiler/template_loader';
 import {CompileStepFactory, DefaultStepFactory} from './compile_step_factory';
 import {Parser} from 'angular2/change_detection';
 import {ShadowDomStrategy} from '../shadow_dom/shadow_dom_strategy';
-import {PropertySetterFactory} from '../view/property_setter_factory';
```
Comment (tbosch) [inline modules/angular2/src/render/dom/compiler/compiler.ts]: Actually delete `PropertySetterFactory` file

File: modules/angular2/test/render/dom/view/proto_view_builder_spec.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        binder.setComponentId('someComponent');
+        expect(() => builder.build())
+            .toThrowError(
+                `Can't bind to 'unknownProperty' since it isn't a know property of the 'some-custom' element and there are no matching directives with a corresponding property`);
+      });
+
+    });
+
+    describe('property normalization', () => {
+      it('should normalize "innerHtml" to "innerHTML" in JS', () => {
```
Comment (tbosch) [inline modules/angular2/test/render/dom/view/proto_view_builder_spec.ts]: Remove `in JS`

**Comment by tbosch** [comment]
Reviewed with @vsavkin in person.

**Comment by pkozlowski-opensource** [comment]
@tbosch was there any special technical reason to kill `[foo.style="exp"`? Or was it just to make the whole system more consistent?

**Comment by tbosch** [comment]
It's still there, but as style.foo. Was it foo.style before?

**Comment by pkozlowski-opensource** [comment]
@tbosch oh, sorry, I just got confused by the breaking changes section in 0a51ccbd68afa6c5ca9252fa2b9568db49833a9c. All in order!

**Pull request closed without merging**